integrations/terraform/ec2-ssh-bastion/main.tf

```diff
@@ -0,0 +1,73 @@
+# Define the Materialize provider
+terraform {
+  required_providers {
+    materialize = {
+      source  = "MaterializeInc/materialize"
+      version = "0.0.4"
+    }
+    # null = {
+    #   source = "hashicorp/null"
+    #   version = "3.2.1"
+    # }
+  }
+}
+
+# Include aws provider
+provider "aws" {
+  region = "us-east-1"
+}
+
+# Include Materialize provider
+provider "materialize" {
+  host     = local.materialize_host
+  username = local.materialize_username
+  password = local.materialize_password
+  port     = 6875
+  database = "materialize"
+}
+
+# Use the materialize ssh module
+module "ssh_bastion" {
+  source  = "MaterializeInc/ec2-ssh-bastion/aws"
+  version = "0.1.0"
+
+  aws_region     = local.aws_region
+  mz_egress_ips  = local.mz_egress_ips
+  vpc_id         = local.vpc_id
+  subnet_id      = local.subnet_id
+  ssh_public_key = local.ssh_public_key
+}
+
+# Create an SSH connection in Materialize
+resource "materialize_connection_ssh_tunnel" "example_ssh_connection" {
+  name        = "ssh_example_connection"
+  schema_name = "public"
+  host        = module.ssh_bastion.ssh_bastion_server.public_ip
+  port        = 22
+  user        = "ubuntu"
+}
+
+# Upload the example_ssh_connection.public_key_1 to the EC2 ssh bastion server
```

Comment:
Rather than have this commented out, could this resource be a conditional on `local.ssh_private_key`. I think something like.
```
count = local.ssh_private_key? 1 : 0
```
I forget if unset variables in Terraform are truthy or not

Reply:
That is a good suggestion!

The main reason why I had to comment this out was that it does not wait for the EC2 instance to be completely ready, so the SSH connection will fail unless you wait a minute or so for the server to be fully up and running.